functional_testing/Openvcloud/ovc_master_hosted/OVC/a_basic/machine_tests.py

```diff
@@ -141,20 +145,28 @@ def test003_create_machine_with_resize(self):
 
             if size['id'] not in range(1, 7):
                 continue
-            
-            self.lg('3- Stop the machine')
-            self.account_owner_api.cloudapi.machines.stop(machineId=machineId)
-
-            machineInfo = self.api.cloudapi.machines.get(machineId=machineId)
-            self.assertEqual(machineInfo['status'], 'HALTED')
-
-            self.lg('Resize the machine with all possible combinations, should succeed')
+            if machine_status == "online":
+                if size['memory'] < selected_size['memory']:
+                    self.lg('resize running machine with memory size less than selected size, should fail')
+                    with self.assertRaises(ApiError) as e:
+                        self.account_owner_api.cloudapi.machines.resize(machineId=machineId, sizeId=size['id'])
+                    self.assertEqual(e.exception.message, '400 Bad Request')
+                    continue
+
+            if machine_status == "offline":
+                self.lg('- Stop the machine')
+                self.account_owner_api.cloudapi.machines.stop(machineId=machineId)
+                machineInfo = self.api.cloudapi.machines.get(machineId=machineId)
+                self.assertEqual(machineInfo['status'], 'HALTED')
+
+            self.lg('- Resize the machine with {} status with all possible combinations, should succeed'.format(machine_status))
             self.account_owner_api.cloudapi.machines.resize(machineId=machineId, sizeId=size['id'])
 
-            self.lg('4- Start the machine')
-            self.account_owner_api.cloudapi.machines.start(machineId=machineId)
+            if machine_status == "offline":                
+                self.lg('- Start the machine')
+                self.account_owner_api.cloudapi.machines.start(machineId=machineId)
             
-            self.lg('Check that the machine is updated')
+            self.lg('- Check that the machine is updated')
```

Comment:
Please make sure that the sizes are changed inside the machine itself. and to save time u can check only 4 RANDOM sizes instead of 8